network.py

```diff
@@ -57,7 +66,8 @@ def __init__(self):
     def run(self):
         while self.running:
             try:
-                self._read_packet()
+                data = self.socket.recv(BUFFER_SIZE)
+                self._read_packet(data)
```

Comment:
This won't catch a `KeyError` if `parsed` doesn't contain `'sender'`, or other required keys

Reply:
this has been done